.github/workflows/deploy-api.yml

```diff
@@ -16,16 +18,12 @@ env:
 jobs:
   deploy:
     runs-on: ubuntu-latest
-    if: >
-      github.ref == 'refs/heads/main' &&
-      secrets.DEPLOY_SSH_KEY &&
-      secrets.DEPLOY_HOST_DNS &&
-      secrets.DEPLOY_USERNAME &&
-      secrets.DEPLOY_TARGET_DIR
+    if: github.ref == 'refs/heads/main' &&
     steps:
       - uses: actions/checkout@v3
       - run: rm -rf .git
       - uses: easingthemes/ssh-deploy@main
+        if: ${{ env.SSH_PRIVATE_KEY && env.REMOTE_USER && env.REMOTE_HOST && env.TARGET }}
```

Comment:
I'm a bit confused by how we check `if:` for a step that also assigns the same exact environment variables that we are testing. I guess that those secrets exist is a prerequisite and we shouldn't fail silently 🤔

Reply:
I think this should use the secrets context instead of the env context. It might work like this but I think the `if` gets evaluated before the rest of the step.

Reply:
So if the `if` evaluates first, that seems to mean we would never run this step. Because the env variables are assigned only after the if.

Reply:
As noted above I was unsure so I ran [a test](https://github.com/assignUser/test-repo-a/actions/runs/7294824144/job/19880310519) and I was wrong. The env context is indeed updated with `env` from the current step. This is good to know :D